.azure-pipelines/build-template.yml

```diff
@@ -0,0 +1,70 @@
+parameters:
+- name: platform
+  type: string
+  values:
+  - broadcom
+  - mellanox
+- name: platform_short
+  type: string
+  values:
+  - brcm
+  - mlnx
+
+- name: cache_mode
+  type: string
+  values:
+  - wcache
+  - rcache
+  - cache
+
+- name: dbg_image
+  type: boolean
+
+- name: swi_image
+  type: boolean
+
+- name: raw_image
+  type: boolean
+
+- name: sync_rpc_image
+  type: boolean
+
+jobs:
+- job:
+  displayName: ${{ parameters.platform }}
+  timeoutInMinutes: 600
+  steps:
+  - template: cleanup.yml
+  - checkout: self
+    submodules: recursive
+    displayName: 'Checkout code'
+  - script: |
```

Comment:
Add clean=true in checkout, so this block can be removed.

Reply:
If fsroot exist, checkout with "clean=true" will fail because of permission.
Put clean steps together. Do cleaning job pre-build and post-build. This is more robust than build-in clean.